### PR TITLE
[spec/interpreter/test] Rename memory instructions

### PR DESCRIPTION
**Pull request by rossberg**
This is a PR to implement #627. It renames:

* `current_memory` -> `mem.size`
* `grow_memory` -> `mem.grow`
* `memory` -> `mem` (in memory declarations, imports, and exports)

It also adjusts the naming conventions in the interpreter and adapts all tests.

### CODE REVIEW

**Comment by jfbastien** [comment]
Can discuss and vote on this in the next CG meeting before doing the change? I don't think it's purely editorial given that some tooling already relies on it.

**Comment by rossberg** [comment]
@jfbastien, absolutely. Didn't plan to land this before that.

File: document/js-api/index.bs [hunk 1]
```diff
@@ -93,10 +93,10 @@ urlPrefix: https://webassembly.github.io/spec/core/; spec: WebAssembly; type: df
     text: trap; url: exec/runtime.html#syntax-trap
     url: exec/runtime.html#values
         text: WebAssembly value
-        text: 𝗂𝟨𝟦.𝖼𝗈𝗇𝗌𝗍
-        text: 𝗂𝟥𝟤.𝖼𝗈𝗇𝗌𝗍
-        text: 𝖿𝟥𝟤.𝖼𝗈𝗇𝗌𝗍
-        text: 𝖿𝟨𝟦.𝖼𝗈𝗇𝗌𝗍
+        text: 헂ퟨퟦ.햼허헇헌헍
```
Comment (binji) [inline document/js-api/index.bs]: o_O

What happened here? :-)

File: interpreter/host/spectest.ml [hunk 2]
```diff
@@ -38,5 +38,5 @@ let lookup name t =
   | "global", ExternGlobalType t -> ExternGlobal (global t)
   | "global", _ -> ExternGlobal (global (GlobalType (I32Type, Immutable)))
   | "table", _ -> ExternTable table
-  | "memory", _ -> ExternMemory memory
+  | "memory", _ -> ExternMem mem
```
Comment (binji) [inline interpreter/host/spectest.ml]: `"mem"` too?

File: document/js-api/index.bs (hunk 1 above)
Comment (rossberg) [reply document/js-api/index.bs]: Wow, no idea. I think my editor barfed on the extended Unicode. Fixed, but @littledan, why this fancy? I for one have no idea how I would even input these characters if I needed to edit any of this.

File: interpreter/host/spectest.ml (hunk 2 above)
Comment (rossberg) [reply interpreter/host/spectest.ml]: Done.

**Comment by ballercat** [comment]
This only affects text encoding correct? All binary encoding and opcodes remain the same, yes?

**Comment by rossberg** [comment]
@ballercat, yes.

**Comment by binji** [comment]
This came up at the Feb 21st CG meeting. There wasn't much discussion, other than some mentioning they didn't like `memory` -> `mem` for the section name.

I see now that all we really needed was a CG vote for this, but that didn't happen unfortunately. Perhaps the best thing to do for now is update this PR and wait for the next one to get an official vote?

**Comment by rossberg** [comment]
Alternative PR with long names: #720.

**Comment by rossberg** [comment]
Abandoned in favour of #720 after CG decision.

**Pull request closed without merging**